File: days_mobile/lib/models/Absence.dart
class Absence {
  String className;
  String absenceDate;

  Absence({this.className, this.absenceDate});

  Absence.fromJson(Map<String, dynamic> json) {
    className = json['class_name'];
    absenceDate = json['absence_date'];
  }

  Map<String, dynamic> toJson() {
    final Map<String, dynamic> data = new Map<String, dynamic>();
    data['class_name'] = this.className;
    data['absence_date'] = this.absenceDate;
    return data;
  }
}

File: days/app/Http/Resources/Absence.php
<?php

namespace App\Http\Resources;

use Illuminate\Http\Resources\Json\JsonResource;

class Absence extends JsonResource
{
    /**
     * Transform the resource into an array.
     *
     * @param  \Illuminate\Http\Request  $request
     * @return array
     */
    public function toArray($request)
    {
        return [
            'class_name' => $this->class->name,
            'absence_date' => $this->absence_date,
        ];
    }
}
